core/src/main/java/org/elasticsearch/action/search/TransportMultiSearchAction.java

```diff
@@ -123,7 +121,8 @@ void executeSearch(
             final Queue<SearchRequestSlot> requests,
             final AtomicArray<MultiSearchResponse.Item> responses,
             final AtomicInteger responseCounter,
-            final ActionListener<MultiSearchResponse> listener) {
+            final ActionListener<MultiSearchResponse> listener,
+            long startTimeInNanos) {
```

Comment:
Please make this parameter final, and name it `relativeStartTime` (to make it clear that it's start time, but it's a relative and not absolute start time).